hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -417,6 +418,7 @@ private boolean initializeFromFilesystem(String dataTableInstantTime, List<Metad
       }
     }
 
+    Lazy<List<Pair<String, FileSlice>>> lazyLatestMergedPartitionFileSliceList = getLazyLatestMergedPartitionFileSliceList();
```

Comment:
can we clear this entry in closeInternal() of the same class. 
So, that on re-initialization,  we don't keep using older merged file slice state.

Reply:
The point is to reuse it across multiple metadata partition types.  The assumption is that during initialization, the data table does not change, and the file system view does not change.  If this assumption does not hold, there is a bigger problem of not maintaining the transaction here.

Reply:
OK. I assume no change is needed now since this is a local variable.

Reply:
LGTM